Binary files emsdk-orig/upstream/emscripten/cache/sysroot/lib/wasm32-emscripten/libwasmfs-mt.a and emsdk/upstream/emscripten/cache/sysroot/lib/wasm32-emscripten/libwasmfs-mt.a differ
diff -urN emsdk-orig/upstream/emscripten/system/lib/wasmfs/wasmfs.cpp emsdk/upstream/emscripten/system/lib/wasmfs/wasmfs.cpp
--- emsdk-orig/upstream/emscripten/system/lib/wasmfs/wasmfs.cpp	2022-07-29 17:22:28.000000000 +0000
+++ emsdk/upstream/emscripten/system/lib/wasmfs/wasmfs.cpp	2022-08-06 02:07:24.098196400 +0000
@@ -141,7 +141,7 @@
     }
 
     auto inserted =
-      lockedParentDir.insertDirectory(childName, S_IRUGO | S_IXUGO);
+      lockedParentDir.insertDirectory(childName, S_IRUGO | S_IWUGO | S_IXUGO);
     assert(inserted && "TODO: handle preload insertion errors");
   }
 
